apps/root/src/pages/position-detail/components/position-summary-controls/index.tsx

```diff
@@ -370,58 +369,55 @@ const PositionSummaryControls = ({ pendingTransaction, position }: PositionSumma
     }
   };
 
-  const options = React.useMemo<OptionsMenuOption[]>(
-    () => [
-      {
-        type: OptionsMenuOptionType.option,
-        label: intl.formatMessage(
-          defineMessage({
-            description: 'withdrawToken',
-            defaultMessage: 'Withdraw {token}',
-          }),
+  const options = [
```

Comment:
This change solves a bug: due to the early return in this function, we had the 'Rendered fewer hooks than expected' error. We could move this array up but I don't think we'd gain too much by memoizing it

Reply:
Mmmm seems like this a solution for a code smell. We would this not be rendered after X action? that makes the error throw?

Reply:
Kinda, This options are only displayed if the connected wallet is position owner. If the user is not connected and then connects the wallet, being the owner of the position, the component re-renders and has this hooks issue.
It's not recommended anyway to have hooks after an early return based on [official docs](https://arc.net/l/quote/ghusmmuw)

I'm removing the early return so we can keep the memo :)

Reply:
Yeah and lets fix this so it doesnt render until we have the user and positions